SegmentEditorDrawTube/SegmentEditorDrawTubeLib/SegmentEditorEffect.py

```diff
@@ -340,21 +340,25 @@ def createNewModelNode(self):
   def createNewMarkupNode(self):
     # Create empty markup fiducial node
     if self.segmentMarkupNode is None:
-      displayNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLMarkupsDisplayNode")
-      displayNode.SetSaveWithScene(False)  # prevent temporary node from being saved into the scene
-      displayNode.SetTextScale(0)
       self.segmentMarkupNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLMarkupsFiducialNode")
       self.segmentMarkupNode.SetSaveWithScene(False)  # prevent temporary node from being saved into the scene
+      displayNode = self.segmentMarkupNode.GetDisplayNode()
```

Comment:
Doesn’t `slicer.mrmlScene.AddNewNodeByClass("vtkMRMLMarkupsFiducialNode")` always create a display node?

Reply:
@lassoan I've pushed an update to add `self.segmentMarkupNode.CreateDefaultDisplayNodes()`

Reply:
> Doesn’t slicer.mrmlScene.AddNewNodeByClass("vtkMRMLMarkupsFiducialNode") always create a display node?

`slicer.mrmlScene.AddNewNodeByClass(...)` just adds a single node. Eventually, some modules or GUI component may decide that the node should be displayed and adds a display node, but if it happens may depend on many things (e.g., if the current layout is a one-up table view then a display node may not be added - just an example, I'm not sure what triggers the display node creation in this case).

Reply:
Sorry for keep asking for more, but could you check if the created display node is saved with the scene? If it is then we should prevent that, because it may result in having an orphan display node, which can cause problems. Solution can be to disable `SaveWithScene` for the display node, too.

Reply:
Yes when I get the created display node for this effect's markup node, `GetSaveWithScene` returns 0/False. Therefore it shouldn't be saved with the scene.

Reply:
@lassoan `GetSaveWithScene` returning 0 is confirmation that it wouldn't be saved with the scene, correct?